### DISCUSSION TITLE
Different between BaseConteroller and FrontendController

### DISCUSSION THREAD

**Discussion by chrobaks** [pimcore/pimcore#16306] [General]
Hi, i just make the Pimcore Acadamy and for that i installed the docker container solution (Demo version) from the Pimcore Documentation. In the Junior Course/Documents -> The Template-Video is a view to the DefaultController extended a FrontendController. But in the Demo version of Pimcore (ver. 11.1.1) the DefaultController extended a BaseController. Is there two different extended Controller, that i can use in this case? Why is the Demo version different to the version i can see in the Learning-Video? (Sorry for my bad english)

**Comment by chrobaks** [comment]
Sorry, i found a part of my question after searching for the FrontendController. Found it in  vendor/pimcore/pimcore/lib/Controller/FrontendController.php.  When i have to use BaseController and when if have to use FrontendController?

**Reply by kingjia90** [reply]
simple answer is to use BaseController when you need these 2 methods in there 😄 
https://github.com/pimcore/demo/blob/11.1/src/Controller/BaseController.php
